QEP-auxiliary-storage.rst

```diff
@@ -0,0 +1,178 @@
+.. _qep#[.#]:
+
+========================================================================
+QGIS Enhancement ??: Auxiliary storage
+========================================================================
+
+:Date: October 2015
+:Author: Hugo Mercier
+:Contact: hugo dot mercier at oslandia dot com
+:Last Edited: 
+:Status:  
+:Version:
+
+Summary
+-------
+
+This proposal is about an evolution of QGIS aiming at storing auxiliary data
+in a layer. Such auxiliary data are data used mostly for the needs of QGIS (symbology) and have no real
+interest in being stored with the native raw geospatial data.
+
+The need arises from the restrictions existing in the manual placement of labels.
+Manual placement of labels are possible in QGIS by setting some labeling properties (X and Y position,
+and rotation angle optionaly) as being "data-defined", meaning that values come from a column (or an expression).
+But setting this up on an existing layer requires either to add new columns to the source layer, while it is
+not always possible or desirable. Creating another table and using joins to it is possible but the synchronisation
+between the two tables is not possible, leading to a solution not satisfactory.
+
+We propose here a more integrated solution where the creation of auxiliary columns or layers
+are brought more transparently to the user by storing them in a central repository carried by the
+project file.
+
+Requirements
+------------
+
+The proposed solution have the following requirements :
+
+- It should allow to transparently add columns of any type to a layer, without adding them to the source data source.
+Beside usual types (string, real, integer), we may also think of geometry types. We only consider types that can be converted to atomic types (string for WKT or WKB)
+The user should be able to edit values in these columns when the layer is in editing mode.
+
+- These values should not be stored with the source data, and should be carried by the project, i.e.
+when the project file is moved or copied, the auxiliary data should still be there.
+
+- Adding a new row to the layer also adds a new row for each auxiliary field (with a null or default value)
```

Comment:
what about only adding the row when data is present in one of the auxiliary fields, and have the default value returned for these fields as null? This would avoid the auxiliary table bloating out with effectively empty rows, and would keep the project file small and responsive even if aux fields are used with a table containing millions of features.